metaflow/plugins/aws/eks/kubernetes_client.py

```diff
@@ -367,31 +387,35 @@ def __repr__(self):
         )
 
     def _fetch_job(self):
+        client = self._client_wrapper.get()
         try:
             return (
-                self._client.BatchV1Api()
+                client.BatchV1Api()
                 .read_namespaced_job(name=self._name, namespace=self._namespace)
                 .to_dict()
             )
-        except self._client.rest.ApiException as e:
+        except client.rest.ApiException as e:
             # TODO: Handle failures as well as the fact that a different
             #       process can delete the job.
             raise e
 
     def _fetch_pod(self):
+        client = self._client_wrapper.get()
         try:
-            # TODO (savin): pods may not appear immediately or they may
-            #               disappear
-            return (
-                self._client.CoreV1Api()
-                .list_namespaced_pod(
-                    namespace=self._namespace,
-                    label_selector="job-name={}".format(self._name),
-                )
-                .to_dict()["items"]
-                or [None]
-            )[0]
-        except self._client.rest.ApiException as e:
+            # Pod objects may not get created immediately after job submission
```

Comment:
Should we do an exponential back-off here instead of constant backoff?

Reply:
I wanted to say yes but after thinking a bit, a constant backoff here should be fine. The rationale for using exponential backoff is usually about retrying service _errors_; since you may be getting the error response because the service is overloaded in the first place, you don't want to hammer it even more with constant retries.

However here we're not retrying an error response, just waiting for the pod to appear.